scripts/generate-test-contracts.sh

```diff
@@ -9,12 +9,34 @@ cp ./contracts/Resolver.sol ./contracts/UpdatedResolver.sol
 cp ./contracts/ColonyNetwork.sol ./contracts/UpdatedColonyNetwork.sol
+cp ./contracts/ReputationMiningCycle.sol ./contracts/UpdatedReputationMiningCycle.sol
+cp ./contracts/IReputationMiningCycle.sol ./contracts/IUpdatedReputationMiningCycle.sol
+# Modify UpdatedColonyNetwork contract
 sed -i.bak "s/contract ColonyNetwork/contract UpdatedColonyNetwork/g" ./contracts/UpdatedColonyNetwork.sol
 sed -i.bak "s/address resolver;/address resolver;function isUpdated() public pure returns(bool) {return true;}/g" ./contracts/UpdatedColonyNetwork.sol
+# Modify UpdatedColony contract
 sed -i.bak "s/contract Colony/contract UpdatedColony/g" ./contracts/UpdatedColony.sol
+sed -i.bak "s/ColonyStorage/UpdatedColonyStorage/g" ./contracts/UpdatedColony.sol
 sed -i.bak "s/function version() public pure returns (uint256 colonyVersion) { return ${version}/function version() public pure returns (uint256 colonyVersion) { return ${updated_version}/g" ./contracts/UpdatedColony.sol
-sed -i.bak "s/contract UpdatedColony is ColonyStorage, PatriciaTreeProofs {/contract UpdatedColony is ColonyStorage, PatriciaTreeProofs {function isUpdated() public pure returns(bool) {return true;}/g" ./contracts/UpdatedColony.sol
+sed -i.bak "s/contract UpdatedColony is UpdatedColonyStorage, PatriciaTreeProofs {/contract UpdatedColony is UpdatedColonyStorage, PatriciaTreeProofs {function isUpdated() public pure returns(bool) {return true;}/g" ./contracts/UpdatedColony.sol
+# Modify UpdatedColonyDataTypes contract
+sed -i.bak "s/ColonyDataTypes/UpdatedColonyDataTypes/g" ./contracts/UpdatedColonyDataTypes.sol
+sed -i.bak "s/mapping (uint8 => mapping (address => uint256)) payouts;/mapping (uint8 => mapping (address => uint256)) payouts; uint256 x;/g" ./contracts/UpdatedColonyDataTypes.sol
+# Modify UpdatedColonyStorage contract
+sed -i.bak "s/ColonyStorage/UpdatedColonyStorage/g" ./contracts/UpdatedColonyStorage.sol
+sed -i.bak "s/ColonyDataTypes/UpdatedColonyDataTypes/g" ./contracts/UpdatedColonyStorage.sol
+# Modify IUpdatedColony contract
 sed -i.bak "s/contract IColony/contract IUpdatedColony/g" ./contracts/IUpdatedColony.sol
-sed -i.bak "s/contract IUpdatedColony is ColonyDataTypes, IRecovery {/contract IUpdatedColony is ColonyDataTypes, IRecovery {function isUpdated() public pure returns(bool);/g" ./contracts/IUpdatedColony.sol
+sed -i.bak "s/ColonyDataTypes/UpdatedColonyDataTypes/g" ./contracts/IUpdatedColony.sol
+sed -i.bak "s/contract IUpdatedColony is UpdatedColonyDataTypes, IRecovery {/contract IUpdatedColony is UpdatedColonyDataTypes, IRecovery {function isUpdated() public pure returns(bool);/g" ./contracts/IUpdatedColony.sol
+# Modify UpdatedReputationMiningCycle contract
+sed -i.bak "s/contract ReputationMiningCycle/contract UpdatedReputationMiningCycle/g" ./contracts/UpdatedReputationMiningCycle.sol
+sed -i.bak "s/WAD;/WAD;function isUpdated() public pure returns(uint256) {return 100;}/g" ./contracts/UpdatedReputationMiningCycle.sol
```

Comment:
Why does this `isUpdated` return `uint256` and the others are booleans? The test still seems to work, which is a little offputting...

Reply:
Yes, indeed. The twist is that the upgrade contracts here start using `forwardTime` meaning the test checking this particular function is skipped.. I will add 2 test runs to solve this, one as it is currently, running under `parity` and another under `ganache-cli` which will include the run of this verification check.